download_from_csv.py

```diff
@@ -1,41 +1,163 @@
+
+    def debug(self, msg):
+        pass
+
+    def warning(self, msg):
+        pass
 
+    def error(self, msg):
+        print(msg)
+
+
+def download_finish(d):
+    if d['status'] == 'finished':
+        print '\x1b[1A\x1b[2K'
+        print "\x1b[1A[\033[93mConverting\033[00m] %s" % d['filename']
+
+
+def download_image(keywords):
+    keywords2 = '+'.join(keywords.split(' '))
+    url = 'https://www.google.co.in/search?q=%s\&tbm=isch' % keywords2
+    response = requests.get(url, headers=headers)
+    htm = response.text
+    soup = BeautifulSoup(htm, 'html.parser')
+    imageURL = soup.find('img').get('src')
+    wget.download(imageURL, out=keywords)  # saves image as name "images"
+
+
+def generate_csv(URI, username, auth_token):
+    cmd = 'curl -X GET "https://api.spotify.com/v1/users/'\
+        + username + '/playlists/' + URI\
+        + '/tracks" -H "Accept: application/json" -H '\
+        + '"Authorization: Bearer ' + auth_token + '"'
+    os.system(cmd + ' > data.json')
+    data = open('data.json', 'r+').read()
+    os.system('rm data.json')
+    data = json.loads(data)
+    csv_data = UserString.MutableString()
+    try:
+        if data['items']:
+            print "Everything is fine"
+    except:
+        raise Exception("Either token is not good, or username or playlist\
+                URI")
+    for track in xrange(len(data['items'])):
+        name = data['items'][track]['track']['name']
+        artist = data['items'][track]['track']['artists'][0]['name']
+        album = data['items'][track]['track']['album']['name']
+        date = data['items'][track]['added_at']
+        csv_data.append(', '.join([name, artist, album, date]) + '\n')
+    csv_data = str(csv_data)
+    try:
+        f = open('playlist.csv', 'w+')
+        f.write(csv_data)
+        f.close()
+    except Exception as e:
+        raise Exception("Couldn't write csv file to disk")
+    return 'playlist.csv'
+
+# Command line parser arguments
 parser = argparse.ArgumentParser()
-parser.add_argument('-f', '--folder', help="keep the files in the folder specified")
-parser.add_argument('-c', '--create', help="try to create folder if doesn't exist",
-                    action="store_true")
-parser.add_argument('--skip', help="number of songs to skip from the start of csv",
-                    type=int)
-parser.add_argument('csv', help="input csv file")
+parser.add_argument('-f', '--folder', default=os.getcwd(),
+                    help="keep the files in the folder specified")
+parser.add_argument('-c', '--create', action="store_true",
+                    help="try to create folder if doesn't exist")
+parser.add_argument('--skip', type=int,
+                    help="number of songs to skip from the start of csv")
+parser.add_argument('-s', '--search', nargs='*',
+                    help="search source for songs, default is youtube")
+parser.add_argument('-p', '--playlist',
+                    help="Enter the playlist URI(with username and token)")
```

Comment:
change the tone of the sentence.
Use "Enter the ..." when you are interactively asking user for something, like in `raw_input()`
For command line arguments, tell what is expected, like 'spotify playlist URI (this will require username and auth token)'
You can check various man pages to get a hang of what I am saying.